### PR TITLE
feat(content-sidebar): Add data-target-id to menu items

### PR DESCRIPTION
**Pull request by jmalinna**
### Description
Add data-target-id attribute to General Task option item
Add data-target-id attribute to Approval Task option item
Add data-target-id attribute to items in modal visible after clicking on one of the above options

### CODE REVIEW

File: src/elements/content-sidebar/AddTaskMenu.js [hunk 1]
```diff
@@ -28,7 +28,11 @@ const AddTaskMenu = (props: Props) => (
             </MenuToggle>
         </Button>
         <Menu className="bcs-AddTaskMenu">
-            <MenuItem className="bcs-AddTaskMenu-menuItem" onClick={() => props.onMenuItemClick(TASK_TYPE_GENERAL)}>
+            <MenuItem
+                className="bcs-AddTaskMenu-menuItem"
+                data-target-id="MenuItem-generaltask"
```
Comment (wpiesiak) [inline src/elements/content-sidebar/AddTaskMenu.js:33]: I noticed that `camelCase` notation is mostly used for the second part of `data-target-id `across the repository. Is there any specific reason why this notation isn't being followed here?

Comment (jmalinna) [reply src/elements/content-sidebar/AddTaskMenu.js:33]: Yes, we use this structure for `data-target-id`. I contacted Kamil and he shared the examples.

Comment (jmalinna) [reply src/elements/content-sidebar/AddTaskMenu.js:33]: While working with Babel plugin I noticed that it uses camel case ids like this: `data-target-id="Button-deleteAssetText"`.
So I'm wondering why the structure Kamil shared doesn't match.
Let me confirm it.

Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/AddTaskMenu.js:33]: I can confirm that with the previous addition of `data-target-id` attr manually, I copied the naming from `data-testid` and it was not camelCase. However I agree that we should use camelCase here and in the future as this is mostly how it's named automatically by Babel plugin.

Comment (jmalinna) [reply src/elements/content-sidebar/AddTaskMenu.js:33]: Thanks for confirming, I'll update it to camel case!

Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/AddTaskMenu.js:33]: Thank you

File: src/elements/content-sidebar/activity-feed/task-form/TaskForm.js [hunk 2]
```diff
@@ -382,7 +382,10 @@ class TaskForm extends React.Component<Props, State> {
                             className={pillSelectorOverlayClasses}
                             error={this.getErrorByFieldname('taskAssignees')}
                             disabled={isForbiddenErrorOnEdit}
-                            inputProps={{ 'data-testid': 'task-form-assignee-input' }}
+                            inputProps={{
+                                'data-testid': 'task-form-assignee-input',
+                                'data-target-id': 'CreateTaskModal-selectAssigneesInput',
```
Comment (tjuanitas) [inline src/elements/content-sidebar/activity-feed/task-form/TaskForm.js:387]: I see from the other comment that the structure is: `[elementName]-[identifier]`

how was `CreateTaskModal` determined as the element name in this case?

Comment (jmalinna) [reply src/elements/content-sidebar/activity-feed/task-form/TaskForm.js:387]: @kkuliczkowski-box said that when we add ids manually then we can create meaningful names.
Wdyt @kkuliczkowski-box?

Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/activity-feed/task-form/TaskForm.js:387]: I also think that using the `elementName` as the first part should be maintained.

**Review by kkuliczkowski-box — APPROVED** [review]

File: src/elements/content-sidebar/AddTaskMenu.js [hunk 3]
```diff
@@ -28,7 +28,11 @@ const AddTaskMenu = (props: Props) => (
             </MenuToggle>
         </Button>
         <Menu className="bcs-AddTaskMenu">
-            <MenuItem className="bcs-AddTaskMenu-menuItem" onClick={() => props.onMenuItemClick(TASK_TYPE_GENERAL)}>
+            <MenuItem
+                className="bcs-AddTaskMenu-menuItem"
+                data-target-id="MenuItem-generalTask"
```
Comment (bfoxx1906) [inline src/elements/content-sidebar/AddTaskMenu.js:33]: In a previous [PR](https://github.com/box/box-ui-elements/pull/3592/files) the part of the id following the - was camelcase but the first letter was capitalized. Shouldn't we be consistent?

Comment (jmalinna) [reply src/elements/content-sidebar/AddTaskMenu.js:33]: Wi stick to how it's named automatically by Babel plugin, example: https://git.dev.box.net/Hubs/hubs-client/pull/944/files
@bfoxx1906

**Pull request merged**